### PR TITLE
Refactor SendTransaction Component

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Description**
There is very similar code spread across 5 components. This PR removes the eip1559 gas polling logic away from the send flow (confirm screen). The TransactionReview and EditGasFee screens is able to calculate the gas transaction cost. This logic is abstracted away in the gas polling file. The changes in this PR shouldn't modify existing behaviour when QA'd.

Progresses
https://github.com/MetaMask/mobile-planning/issues/272

**Screenshots/Recordings**

https://user-images.githubusercontent.com/29962968/184000489-af5061ad-f713-4f79-bd0a-33e018eb475b.mov

**Issue**

Progresses https://github.com/metamask/mobile-planning/issues/269

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/Views/SendFlow/Confirm/index.js [hunk 1]
```diff
@@ -1062,24 +919,27 @@ class Confirm extends PureComponent {
...
-    this.setState({
-      EIP1559TransactionData: { ...this.state.EIP1559TransactionDataTemp },
-      LegacyTransactionData: { ...this.state.LegacyTransactionDataTemp },
-      gasSelected,
-      gasSelectedTemp: gasSelected,
-      advancedGasInserted: !gasSelected,
-      stopUpdateGas: false,
-    });
+  saveGasEdition = (gasTxn) => {
+    // add error handling and set error state
```
Comment (sethkfman) [inline app/components/Views/SendFlow/Confirm/index.js:923]: remove the commented code if it is not needed

File: app/components/UI/TransactionReview/TransactionReviewEIP1559Update/index.tsx [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+    renderableGasFeeMaxConversion,
+    timeEstimateColor,
+    timeEstimate,
+    timeEstimateId,
+  } = gasTransaction;
+
+  const openLinkAboutGas = useCallback(
+    () =>
+      Linking.openURL(
+        'https://community.metamask.io/t/what-is-gas-why-do-transactions-take-so-long/3172',
```
Comment (sethkfman) [inline app/components/UI/TransactionReview/TransactionReviewEIP1559Update/index.tsx:91]: move URL to url constants folder

File: app/components/UI/TransactionReview/TransactionReviewEIP1559Update/index.tsx [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+  const styles = createStyles(colors);
+
+  const gasTransaction = useGasTransaction({
+    onlyGas: undefined,
+    gasSelected,
+    legacy: !!legacy,
+    gasLimit: null,
+  });
+
+  //   console.log('gasTransaction', selectedGasValue);
```
Comment (sethkfman) [inline app/components/UI/TransactionReview/TransactionReviewEIP1559Update/index.tsx:73]: remove comment

File: app/components/Views/SendFlow/Confirm/index.js [hunk 4]
```diff
@@ -1062,24 +919,27 @@ class Confirm extends PureComponent {
...
+    // });
+    // const error =
+    //   this.state.EIP1559TransactionDataTemp.error ||
+    //   this.state.LegacyTransactionDataTemp.error;
+    // this.setError(error);
 
     this.review();
   };
 
+  existingGas = {
```
Comment (sethkfman) [inline app/components/Views/SendFlow/Confirm/index.js:937]: Where are these values coming from?

File: app/components/Views/SendFlow/Confirm/index.js [hunk 5]
```diff
@@ -1532,31 +1357,16 @@ class Confirm extends PureComponent {
...
-              totalMaxNative={EIP1559TransactionData.renderableTotalMaxNative}
-              gasFeeNative={EIP1559TransactionData.renderableGasFeeMinNative}
-              gasFeeConversion={
-                EIP1559TransactionData.renderableGasFeeMinConversion
-              }
-              gasFeeMaxNative={EIP1559TransactionData.renderableGasFeeMaxNative}
-              gasFeeMaxConversion={
-                EIP1559TransactionData.renderableGasFeeMaxConversion
-              }
+            <TransactionReviewEIP1559Update
```
Comment (sethkfman) [inline app/components/Views/SendFlow/Confirm/index.js:1360]: Make sure to change these component names when you delete the old ones.

File: app/util/general/index.js [hunk 6]
```diff
@@ -80,3 +80,8 @@ export const renderShortText = (text, chars = 4) => {
     return text;
   }
 };
+
+export const chooseOption = (option1, option2) => {
```
Comment (sethkfman) [inline app/util/general/index.js:84]: Add comments & better named variables

File: app/components/UI/EditGasFee1559Update/index.tsx [hunk 7]
```diff
@@ -0,0 +1,771 @@
...
+import { useAppThemeFromContext, mockTheme } from '../../../util/theme';
+import { EditGasFee1559UpdateProps } from './types';
+import createStyles from './styles';
+
+const GAS_LIMIT_INCREMENT = new BigNumber(1000);
+const GAS_INCREMENT = new BigNumber(1);
+const GAS_LIMIT_MIN = new BigNumber(21000);
+const GAS_MIN = new BigNumber(0);
+
+const EditGasFee1559Update = ({
```
Comment (sethkfman) [inline app/components/UI/EditGasFee1559Update/index.tsx:34]: Same comment in previous PRs. EditGasFee1559UpdateProps -> EditGasFee1559Props

**Review by sethkfman — COMMENTED** [review]
Minor comments

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by sethkfman** [comment]
@blackdevelopa Please resolve conflicts

**Comment by chrisleewilcox** [comment]
@blackdevelopa not able to send on Binance testnet.  Getting JSON-RPC error...
https://recordit.co/HCLAxi2VPp

**Comment by chrisleewilcox** [comment]
Send refactor looks good on the following testnets....

- Goerli (ETH), priority level low
- Fuji (AVAX), priority level low
- Mumbai (MATIC), use max, priority level low

**Comment by blackdevelopa** [comment]
> @blackdevelopa not able to send on Binance testnet. Getting JSON-RPC error... https://recordit.co/HCLAxi2VPp

resolved. http://recordit.co/1ZkJUCgmb0

**Comment by chrisleewilcox** [comment]
Send is good on BNB testnet now.

**Pull request merged**